lib/machiiro/ruby/support/core_ext/date.rb

```diff
@@ -31,7 +31,36 @@ def self.parse_period(period)
     Date.new(year, month, 1)
   end
 
+  # Returns the date parsed from the target, otherwise returns nil
+  # @return [Date, nil]
+  def self.parse_period_safe(period)
+    return nil unless period_parsable? period
+    parse_period period
+  end
+
+  # Returns the date parsed from the target, otherwise raises an error
+  # @raise [ArgumentError] if the target is not a string with 6 digits
+  # @return [Date]
+  def self.parse_period!(period)
+    raise ArgumentError, '"period" should can be a string with 6 digits' unless period_parsable? period
+    parse_period period
+  end
+
+  # Returns the date parsed from the target, otherwise returns a period based on the current date
+  # @return [Date]
+  def self.parse_period_or_now(period)
+    target = period_parsable?(period) ? period : Time.zone.now.to_ym_i
+    parse_period target
+  end
+
   def self.today_at_jst
     Time.now.to_jst.to_date
   end
+
+  # Returns true if the target can be a number with 6 digits.
+  # NOTE: This method does not check the format of the target.
+  def self.period_parsable?(target)
+    return false if target.nil?
+    target.to_s =~ /\A\d{6}\z/ ? true : false
```

Comment:
可読性がややイマイチなのですが、ホットスポットになったとしてもボトルネックにならないよう速さを優先した実装にしました。また、 `to_s` メソッドは常に反応するものとして、チェックなどを挟んでいません。

簡単ですがベンチマークを取りました。
```ruby
User = Struct.new(:name, :first_name, :last_name, :age, :address1, :address2, :tel, keyword_init: true)

def new_random_user
  User.new(
    name: Faker::Name.name,
    first_name: Faker::Name.first_name,
    last_name: Faker::Name.last_name,
    age: Faker::Number.between(from: 18, to: 65),
    address1: Faker::Address.street_address,
    address2: Faker::Address.secondary_address,
  )
end

def new_random_period
  random_month = Faker::Number.between(from: 1, to: 12)
  "2025#{"%02d" % random_month}".to_i
end

RANDOMIZED_ELEMENTS = [-> { new_random_user }, -> { new_random_period }, -> { nil }]

random_elements = []
p 'Start to generate random elements'
200_000.times do
  random_elements << RANDOMIZED_ELEMENTS.sample.call
end

def period_parsable?(target)
  return false if target.nil?
  target.to_s =~ /\A\d{6}\z/ ? true : false
end


p 'Start to check random elements'
Benchmark.bmbm do |reporter|
  reporter.report('random_elements') do
    random_elements.each do |element|
      period_parsable?(element)
    end
  end
end


# "Start to generate random elements"
# "Start to check random elements"
# Rehearsal ---------------------------------------------------
# random_elements   0.178395   0.001494   0.179889 (  0.180177)
# ------------------------------------------ total: 0.179889sec

#                       user     system      total        real
# random_elements   0.172502   0.001519   0.174021 (  0.174357)

```

Reply:
良いと思います 👍 細かいですが正規表現で boolean を返すだけであれば `match?` がおすすめです
```suggestion
    target.to_s.match?(/\A\d{6}\z/)
```